docs/cpan-dependency-confusion.md

```diff
@@ -0,0 +1,247 @@
+
+Specifying the requirements in Makefile.PL, Build.PL or META.json/META.yml achieves the same and was used in some cases (e.g. CPAN/CPANPLUS)
+
+On DarkPAN:
+```
+PrivateAndPublicGreaterCPAN@0.03
+Private@0.01
+PrivateAndPublicGreaterLocal@0.02
+```
+
+On CPAN:
+```
+PrivateAndPublicGreaterCPAN@0.04
+Public@0.12
+PrivateAndPublicGreaterLocal@0.01
+```
+
+`Private` is only present on DarkPAN, `Public` is only present on CPAN.
+
+`PrivateAndPublicGreaterCPAN` and `PrivateAndPublicGreaterLocal` are requirements that can be "confused" (if install ends up with `PrivateAndPublicGreaterCPAN@0.04` and/or `PrivateAndPublicGreaterLocal@0.02`) 
+
+GrayPAN/DarkPAN was simulated with the following tools:
+- [pinto](https://metacpan.org/dist/Pinto) either targetted via HTTP `http://0.0.0.0:8000/` or `file:///data/darkpan` for `cpanm`, `cpm`, `carton`, `carmel` and `cpan`
+- [CPAN::Mini](https://metacpan.org/dist/CPAN-Mini) for `cpanplus` (cpanplus index fetch incompatible with pinto)
+- [StratoPAN](https://stratopan.com/)/[JFrog Artifactory](https://jfrog.com/artifactory/) for CPAN artifact manager proxy
+
+## Dependency confusion when local is non-existent
+### Unsafe
+- Specifying the private requirements in Makefile.PL, Build.PL, META.* or cpanfile/cpmfile is unsafe. Doing manual installs (via EUMM/MB or `cpanm tarball`) and then relying on requirements being already satisfied (and CPAN client not being "zealous") to not install anything public is risky, to say the least.
+
+### Safe
+- Do not specify private requirements in Makefile.PL, Build.PL, META.* or cpanfile/cpmfile. Specifying requirements is technically not required for runtime. This is suboptimal because it makes dependencies implicit and shifts right the verification of requirements.
+- Use reserved namespaces (`_Underscore::Trick` or `UN_DESCORE::Trick`) that can not resolve to public modules.
+
+### Documented but unsafe
+`Local::*` is documented as a name space for private modules but as of November 2024 only reserved "by convention", see [Do not index Local::* (nor Local)](https://github.com/andk/pause/pull/541)
+
+## Dependency version confusion between PANs
+### Unsafe
+#### cpanm --mirror
+- It keeps resolving first via public indexer (has no information about DarkPAN versions)
+- It only "appends" so beware mirrors installed first via `PERL_CPANM_OPT`
+- Order is important
+
+e.g.:
+- Problem with "appends": `PERL_CPANM_OPT="--mirror https://cpan.org" cpanm --mirror-only --mirror http://darkpan.local --installdeps .`
+- Problem with "indexer": `cpanm --mirror http://darkpan.local --mirror https://cpan.org --installdeps .`
+- Problem with "ordering": `cpanm --mirror-only --mirror https://cpan.org --mirror http://darkpan.local --installdeps .`
+
+#### cpanfile/cpanm
+- `mirror ""` and `mirror =>` directives from cpanfile are (silently) ignored by cpanm (!)
```

Comment:
Is this true? A bit surprising, actually... Would it be meaningful to link to an issue related to this?